packages/headless/src/features/facets/facet-set/facet-set-actions.ts

```diff
@@ -60,7 +61,7 @@ export interface RegisterFacetActionCreatorPayload {
    *
    * @defaultValue `automatic`
    */
-  sortCriteria?: FacetSortCriterion;
+  sortCriteria?: FacetSortCriterion | SpecificSortCriteriaExplicitAlphanumeric;
```

Comment:
To re-emphasize on my above point, another way to look at it is that everything "external facing" (controllers, actions) should not even expose anything regarding that type of sort criteria. It adds confusion/complexity to have something polymorphic for sort (ie: 2 different ways to specify sort, even if the API does support it).

The only place that new type should be added is at the "building the request that get sent to the search API" level, where we detect that what's in the store is the string `alphanumericDescending`, and build the proper object with `type` / `order`.

Reply:
This makes sense. So just to confirm, _**all instances of sortCriteria**_ should be of type `FacetSortCriterion` _**except for in the `FacetRequest` interface**_, in which case `SpecificSortCriteriaExplicitAlphanumeric` is also allowed?

If this is the case, I'll update the logic here such that there is no mention of the FacetRequest type, but `buildFacetRequest` will still "quietly" update sortCriteria to accommodate the new Search API specs.

Does this sound good to you? 🙂